service/worker/scanner/taskqueue/db.go

```diff
@@ -95,6 +97,7 @@ func (s *Scavenger) listTaskQueue(
 			PageSize:  pageSize,
 			PageToken: pageToken,
 		})
+		// todo: it looks like it shadows the err variable from the outer scope
 		return err
```

Comment:
It's actually fine as long as `retryForever` is implemented intuitively: meaning it should try at least once and always return the error from the last attempt (even if it was nil), potentially wrapped. If so that's the last thing that err will get assigned to (on line 94), so the error we return will always be the error from the last attempt.